ENIGMAsystem/SHELL/Graphics_Systems/OpenGL-Common/shader.cpp

```diff
@@ -620,7 +618,7 @@ bool glsl_program_validate(int id)
   if (validated){
     return true;
   } else {
-    std::cout << "Shader program[" << id << "] - Validation failed - Info log: " << std::endl;
+    DEBUG_MESSAGE("Shader program[" + std::to_string(id) + "] - Validation failed - Info log:", MESSAGE_TYPE::M_ERROR);
```

Comment:
Yeah, I am not so sure about this one. If this is a user function, we should just be returning the validation log to the user so they can display it how they want. Now, for all internal validation we do of shaders, including our own shaders, we should log that in debug mode. But I don't think we should be doing it here for when the user is requesting the validation log.

Sorry, it's been a while since I've even really looked at these shader sources. I just started them and Harri basically made them great again.